pkg/installation/installation.go

```diff
@@ -696,3 +681,16 @@ func (i *Installation) releaseSrcFile(path string) string {
 func (i *Installation) releaseFile(path string) string {
 	return fmt.Sprintf(releaseResourcePattern, i.Options.configVersion, path)
 }
+
+func (i *Installation) getHelmSecret(interval time.Duration,retries int) (*corev1.Secret, error){
```

Comment:
It would be better to use some generic retry package, and then call `getHelmSecret` method, wrapping it with the retry function. Then, the `getHelmSecret` would get the secret only once. Something like this:

```go
var secret *corev1.Secret
retry.Do(func() error {
               secret, err = i.getHelmSecret
	}, retry.Attempts(s.retryAttempts), retry.Delay(retryDelayMilliseconds*time.Millisecond))
```

In this case, the `"github.com/avast/retry-go"` package is used.